File: src/maasserver/static/js/angular/factories/staticroutes.js
/* Copyright 2016 Canonical Ltd.  This software is licensed under the
 * GNU Affero General Public License version 3 (see the file LICENSE).
 *
 * MAAS StaticRoute Manager
 *
 * Manages all of the StaticRoutes in the browser. The manager uses the
 * RegionConnection to load the StaticRoutes, update the StaticRoutes, and
 * listen for notification events about StaticRoutes.
 */

function StaticRoutesManager(RegionConnection, Manager) {
  function StaticRoutesManager() {
    Manager.call(this);

    this._pk = "id";
    this._handler = "staticroute";

    // Listen for notify events for the staticroute object.
    var self = this;
    RegionConnection.registerNotifier("staticroute", function(action, data) {
      self.onNotify(action, data);
    });
  }

  StaticRoutesManager.prototype = new Manager();

  return new StaticRoutesManager();
}

StaticRoutesManager.$inject = ["RegionConnection", "Manager"];

export default StaticRoutesManager;

File: src/maasservicelayer/services/staticroutes.py
# Copyright 2024-2025 Canonical Ltd.  This software is licensed under the
# GNU Affero General Public License version 3 (see the file LICENSE).

from maasservicelayer.context import Context
from maasservicelayer.db.repositories.staticroutes import (
    StaticRoutesRepository,
)
from maasservicelayer.models.staticroutes import (
    StaticRoute,
    StaticRouteBuilder,
)
from maasservicelayer.services.base import BaseService


class StaticRoutesService(
    BaseService[StaticRoute, StaticRoutesRepository, StaticRouteBuilder]
):
    def __init__(
        self, context: Context, staticroutes_repository: StaticRoutesRepository
    ):
        super().__init__(context, staticroutes_repository)
